tests/prague/eip7692_eof_v1/eip7069_returndatacopy/test_returndatacopy_execution.py

```diff
@@ -0,0 +1,141 @@
+"""
+RETURNDATACOPY tests covering changes in behavior in EOF.
+"""
+import itertools
+
+import pytest
+
+from ethereum_test_tools import Account, StateTestFiller
+from ethereum_test_tools.common.base_types import Address
+from ethereum_test_tools.common.constants import TestAddress
+from ethereum_test_tools.common.types import Environment, Transaction
+from ethereum_test_tools.eof.v1 import Container, Section
+from ethereum_test_tools.vm.opcode import Opcodes as Op
+
+from .. import EOF_FORK_NAME
+from .spec import CALL_FAILURE, CALL_SUCCESS, EXTCALL_FAILED, EXTCALL_SUCCESS
+
+REFERENCE_SPEC_GIT_PATH = "EIPS/eip-7069.md"
+REFERENCE_SPEC_VERSION = "73f68e76dd92486d13e27375cbb468a2be445a9f"
+
+pytestmark = pytest.mark.valid_from(EOF_FORK_NAME)
+
+_slot = itertools.count(1)
+slot_eof_target_call_status = next(_slot)
+slot_legacy_target_call_status = next(_slot)
+slot_eof_target_returndata = next(_slot)
+slot_eof_target_returndatasize = next(_slot)
+slot_legacy_target_returndatasize = next(_slot)
+
+_address_allocation = itertools.count(0x10000)
+address_entry_point = Address(next(_address_allocation))
+address_callee_eof = Address(next(_address_allocation))
+address_callee_legacy = Address(next(_address_allocation))
+
+pre_storage = {
+    slot_eof_target_call_status: "0xdd",
+    slot_eof_target_returndata: "0xdd",
+    slot_eof_target_returndatasize: "0xdd",
+    slot_legacy_target_call_status: "0xdd",
+    slot_legacy_target_returndatasize: "0xdd",
+}
+
+# Both codes below make an OOB (out-of-bounds) RETURNDATACOPY of one byte,
+# which they then attempt to return (Legacy should exceptionally halt on RETURNDATACOPY).
+code_eof_oob_returndatacopy = Container(
+    sections=[
+        Section.Code(
+            code=Op.RETURNDATACOPY(0, 0, 1) + Op.RETURN(0, 1),
+            max_stack_height=3,
+        )
+    ]
+)
+code_legacy_oob_returndatacopy = Op.RETURNDATACOPY(0, 0, 1) + Op.RETURN(0, 1)
+
+
+@pytest.mark.parametrize("top_level", ("LegacyContract", "EOFContract"), ids=lambda x: x)
+def test_returndatacopy_oob(
+    state_test: StateTestFiller,
+    top_level: str,
+):
+    """Test RETURNDATACOPY correct oob behavior"""
```

Comment:
Could we expand a bit on this docstring?

I think it would be useful to explain a bit more the reason of this test as well as the reasoning behind each parameter.